poseidon/src/lib.rs

```diff
@@ -30,6 +30,10 @@ where
     F: Field,
     MDS: MDSPermutation<F, WIDTH>,
 {
+    /// Create a new Poseidon
```

Comment:
I'm not quite sure about this description..

I just needed something, because clippy wanted documentation with a `# Panics` section.